### PR TITLE
[terra-menu] Fixes Title Click Navigation issue

### PR DESCRIPTION
**Pull request by ST063655**
### Summary

fix for "Terra-Menu" to navigate back on clicking on a submenu back button.
**Jira Link** - https://jira2.cerner.com/browse/UXPLATFORM-8646

Closes #

### Deployment Link

https://terra-framework-deployed-pr-#.herokuapp.com/

### Testing

Code changes are tested with existing WDIO and Jest Test to ensure code changes remain passive.
Tested the description and focus issue via Voice Over and Jaws.

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-menu/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Changed 
+  * fix for "Terra-Menu" to navigate back on clicking on a submenu back button.
+
```
Comment (sdadn) [inline packages/terra-menu/CHANGELOG.md:7]: We can put fixes under its own section
```suggestion
* Fixed 
  * Fix for "Terra-Menu" to navigate back when clicking on a submenu back button.

```

Comment (ST063655) [reply packages/terra-menu/CHANGELOG.md]: Fixed.

File: packages/terra-menu/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Fixed 
+  * Fix for "Terra-Menu" to navigate back when clicking on a submenu back button.
```
Comment (supreethmr) [inline packages/terra-menu/CHANGELOG.md:6]:
```suggestion
  * Fixed issue of navigation on menu header click.
```

**Comment by supreethmr** [comment]
PR description should be updated with how these changes are tested !!

**Review by supreethmr — APPROVED** [review]

File: packages/terra-menu/src/_MenuContent.jsx [hunk 3]
```diff
@@ -292,20 +294,22 @@ class MenuContent extends React.Component {
...
+              onClick={this.props.onRequestBack}
+              onKeyDown={this.onKeyDownBackButton}
+              tabIndex="0"
+              aria-label={backBtnText}
+              aria-describedby={menuHeaderId}
+            >
               {backIcon}
             </div>
-            <h1 className={cx('header-title')}>{this.props.title}</h1>
+            <h1 id={menuHeaderId} className={cx('header-title')}>{this.props.title}</h1>
```
Comment (saket2403) [inline packages/terra-menu/src/_MenuContent.jsx:310]: Can you check with rahul once if h1 for the menu header is valid or not?

Comment (ST063655) [reply packages/terra-menu/src/_MenuContent.jsx]: Changed the header from h1 to h2.

**Review by saket2403 — APPROVED** [review]

**Pull request merged**